### PR TITLE
[QNN EP] Python script to modify Onnx model to make it aligned with converted QNN model

### PR DESCRIPTION
**Pull request by HectorSVC**
Python script to modify Onnx model to make it aligned with converted QNN model

### Description
Onnxruntime QNN EP can support context binary file generated by QNN tool chain. However QNN generated context binary file uses channel last and 8 bits or 16 bits for input and output.
This script get the QNN model input & output information from QNN converted model_net.json file, and insert Cast, Transpose nodes to Onnx model if required.

### CODE REVIEW

**Comment by jywu-msft** [comment]
fix the lint warnings?

**Comment by jywu-msft** [comment]
we don't normally put any python files under core/providers
should we move it to onnxruntime/python/tools/qnn instead?

File: onnxruntime/python/tools/qnn/add_trans_cast.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+def parse_dims(line_content):
+    value = line_content.split("=")
+    assert (len(value) == 2), "Error: Wrong format encountered."
+    value = value[1].replace(" ", "").split(";")
+    assert (len(value) == 2), "Error: Wrong format encountered."
+    shape = value[0].replace("{", "").replace("}", "").split(",")
+    return [eval(item) for item in shape]
+
+
+def parse_qnn_cpp_file(qnn_cpp_file, qnn_input_output_tensor_dic):
```
Comment (adrianlizarraga) [inline onnxruntime/python/tools/qnn/add_trans_cast.py:114]: Can we only parse the model.cpp file? The qnn-onnx-converter also outputs a model_net.json file. Getting this information from the generated model_net.json file would probably be simpler and more reliable.

`model_json["graph"]["tensors"]` is a dictionary of tensor information that includes type, data_type, dims, etc.

Comment (HectorSVC) [reply onnxruntime/python/tools/qnn/add_trans_cast.py:119]: You are right. I didn't notice they output a json file now. That will make it a lot easier and reliable.

---
In reply to: [1235594519](https://github.com/microsoft/onnxruntime/pull/16423#discussion_r1235594519) [](ancestors = 1235594519)

File: onnxruntime/python/tools/qnn/add_trans_cast.py [hunk 2]
```diff
@@ -55,25 +56,28 @@ def graph_topological_sort(graph):
...
+    assert len(value) == 2, "Error: Wrong format encountered."
     value = value[1].split(",")
-    assert (len(value) == 2), "Error: Wrong format encountered."
-    return value[0].replace(" ", "").replace("\"", "")
+    assert len(value) == 2, "Error: Wrong format encountered."
+    return value[0].replace(" ", "").replace('"', "")
 
-class qnn_tensor_struct:
+
+class QNNTENSORSTRUCT:
```
Comment (justinchuby) [inline onnxruntime/python/tools/qnn/add_trans_cast.py:72]: Note: prefer PascalCase for classes?

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**